dataStructures/queues/queue.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"fmt"
+)
+
+// Queue data structure, it will be a list of Orders
+type Queue []*Order
+
+// Order data type
+type Order struct {
+	priority     int
+	quantity     int
+	product      string
+	customerName string
+}
+
+// New method creates a new order with required properties
+func (order *Order) New(priority, quantity int, product, customerName string) {
+	order.priority = priority
+	order.quantity = quantity
+	order.product = product
+	order.customerName = customerName
+}
+
+// Add method takes an order and adds it to the queue based on priority
+func (queue *Queue) Add(order *Order) {
+	if len(*queue) == 0 {
+		*queue = append(*queue, order)
+	} else {
+		appended := false
+
+		for i, existingOrder := range *queue {
+			if order.priority > existingOrder.priority {
+				// since order has a higher priority than the existing order
+				// make a slice till the index
+				initialQueue := (*queue)[:i]
```

Comment:
Variable names in Go should be as short as possible, but not too short (yes, that sounds a bit silly but I'll explain). [More info](https://github.com/golang/go/wiki/CodeReviewComments#variable-names).

```
start := (*queue)[:i]
end := (*queue)[i:]
```

How long / descriptive a variable name should be is relative to how "far away" it is used. Since you're only using these over a few lines they can be relatively short. I removed the `Queue` suffix on the names because you can tell from the `(*queue)` part that it's a Queue. I didn't go as short as `s` and `e` because it's not _immediately_ obvious that they're the start and end of the queue (you can figure it out by looking closely at the index but code should be kind to the reader).